util/ubuntu-dependencies.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+
+# Abort on any error.
+set -e
+
+# This script installs Streisand builder dependencies on an Ubuntu
+# 16.04 system; Debian jessie has been lightly tested. It installs
+# system-wide packages. As a result, this is most useful on a fresh
+# machine, where conflicting Python packages won't be installed. If
+# you're not on a fresh regular machine, consider using
+# venv-dependencies.sh instead.
+#
+# It's safe to run this script multiple times.
+
+# We default to a regular, interactive upgrade:
+quiet=
+# quiet='--yes'
+
+# If you *really* want no interaction (for example, overwriting local
+# config files without prompting), then uncomment this:
+# export DEBIAN_FRONTEND=noninteractive
+
+# If you'd prefer to install dependencies in ~/.local/bin, uncomment
+# this function and comment the next.
+
+#function our_pip_install () {
+#    pip install --user --upgrade "$@"
+#}
+
+function our_pip_install () {
+    sudo -H pip install --upgrade "$@"
+}
+
+### No options below this line. ###
+
+sudo apt-get update
+sudo apt-get $quiet upgrade
+
+# Prefer binaries distributed by upstream OS over those in the pip
+# repository.
+
+# We explicitly want word splitting.
+# shellcheck disable=SC2046
+sudo apt-get $quiet install $(cat <<EOF
+build-essential
+python-dev
+python-setuptools
+python-pip
+python-cffi libffi-dev
+EOF
+)
+
+# Debian doesn't have python-nacl. We'll have to accept the pip
+# version.
+if ! sudo apt-get $quiet install python-nacl libssl-dev; then
```

Comment:
I remember flagging this earlier and suggesting that it would be cleaner to detect Debian from the `lsb_release` output (or something equivalent) and taking the required steps instead of trying one thing and if it fails doing another. Did you have a reason for avoiding that I'm forgetting? 

I'm going to approve & merge this either way in the effort of keeping this moving.

Reply:
Checking for `/etc/debian_version` tells me whether `apt` is effectively available. `lsb_version` seems too precise; perhaps Debian will ship `python-nacl` at some point. I'm also not sure if any Debian-derivatives have funny `lsb_release` results but do have `python-nacl`. 

Yeah, all of this is a flail, but I don't have a controlled environment.  The principled thing would be to have a big nested case statement to handle all tested OS versions, and then fall back to heuristics. 

We could ship `/opt` `.deb`s for various releases, declaring actual dependencies. I'd much rather work on the Docker builder than go through that exercise.

Reply:
Anyway, if you're not convinced, I'll write the case statement for just Debian and Ubuntu, which is easier than trying to figure out why I'm right. :‐)